tests/test_workflow_pipeline_store_install.py

```diff
@@ -0,0 +1,185 @@
+"""Tests for PipelineInstallWorkflow class."""
+
+from contextlib import nullcontext
+from pathlib import Path
+
+import pytest
+import pytest_mock
+
+from nipoppy.config.main import Config
+from nipoppy.config.pipeline import BasePipelineConfig, ProcPipelineConfig
+from nipoppy.env import PipelineTypeEnum
+from nipoppy.layout import DatasetLayout
+from nipoppy.workflows.pipeline_store.install import PipelineInstallWorkflow
+
+from .conftest import create_pipeline_config_files, get_config
+
+
+@pytest.fixture(scope="function")
+def pipeline_config():
+    return ProcPipelineConfig(
+        **{
+            "NAME": "my_pipeline",
+            "VERSION": "1.0.0",
+            "PIPELINE_TYPE": PipelineTypeEnum.PROCESSING,
+        }
+    )
+
+
+@pytest.fixture(scope="function")
+def workflow(tmp_path: Path, pipeline_config: ProcPipelineConfig):
+    dpath_root = tmp_path / "my_dataset"
+    create_pipeline_config_files(
+        tmp_path,
+        proc_pipelines=[pipeline_config.model_dump()],
+    )
+    workflow = PipelineInstallWorkflow(
+        dpath_root=dpath_root,
+        dpath_pipeline=tmp_path
+        / DatasetLayout.pipeline_type_to_dname_map[PipelineTypeEnum.PROCESSING]
+        / "my_pipeline-1.0.0",
+    )
+    # make the default config have a path placeholder string
+    get_config(dicom_dir_map_file="[[NIPOPPY_DPATH_ROOT]]/my_file.tsv").save(
+        workflow.layout.fpath_config
+    )
+    return workflow
+
+
+def _check_files_copied(dpath_source, dpath_dest):
```

Comment:
Renaming to `_assert_files_copied` would make it more obvious that it is performing assertion.